src/main/java/in/projecteka/consentmanager/consent/ConsentScheduler.java

```diff
@@ -0,0 +1,152 @@
+                .switchIfEmpty(Mono.error(ClientError.consentArtefactForbidden()))
+                .filter(this::isGrantedConsent)
+                .switchIfEmpty(Mono.error(ClientError.consentNotGranted()));
+    }
+
+    private Mono<ConsentRepresentation> getConsentWithRequest(String consentId) {
+        return consentArtefactRepository.getConsentWithRequest(consentId)
+                .switchIfEmpty(Mono.error(ClientError.consentArtefactNotFound()));
+    }
+
+    private static boolean isNotSameRequester(ConsentArtefact consentDetail, String requesterId) {
+        return !consentDetail.getHiu().getId().equals(requesterId) &&
+                !consentDetail.getPatient().getId().equals(requesterId);
+    }
+
+    private boolean isGrantedConsent(ConsentRepresentation consentRepresentation) {
+        return consentRepresentation.getStatus().equals(GRANTED);
+    }
+
+    private void updateAndBroadcastConsentExpiry(String consentId, ConsentRepresentation consentRepresentation, ConsentRequestDetail consentRequestDetail) {
+        consentArtefactRepository.updateConsentArtefactStatus(consentId, EXPIRED).block();
+        var hipConsentArtefactRepresentations = getHIPConsentArtefacts(consentRepresentation);
+        hipConsentArtefactRepresentations.forEach(hipConsentArtefactRepresentation ->
+                broadcastConsentArtefacts(hipConsentArtefactRepresentations,
+                        consentRequestDetail.getConsentNotificationUrl(),
+                        consentRepresentation.getConsentRequestId(),
+                        consentRepresentation.getDateModified()).block());
+    }
+
+    private List<HIPConsentArtefactRepresentation> getHIPConsentArtefacts(ConsentRepresentation consentRepresentation) {
+        List<HIPConsentArtefactRepresentation> hipList = new ArrayList<>();
+        hipList.add(from(consentRepresentation.getConsentDetail()));
+
+        return hipList;
+    }
+
+    private HIPConsentArtefactRepresentation from(ConsentArtefact consentArtefact) {
+        HIPConsentArtefact hipConsentArtefact = HIPConsentArtefact.builder()
+                .consentId(consentArtefact.getConsentId())
+                .createdAt(consentArtefact.getCreatedAt())
+                .purpose(consentArtefact.getPurpose())
+                .careContexts(consentArtefact.getCareContexts())
+                .patient(consentArtefact.getPatient())
+                .hip(consentArtefact.getHip())
+                .hiTypes(consentArtefact.getHiTypes())
+                .permission(consentArtefact.getPermission())
+                .consentManager(getConsentManagerRef())
+                .build();
+        String signature = signConsentArtefact(hipConsentArtefact);
+
+        return HIPConsentArtefactRepresentation
+                .builder()
+                .consentDetail(hipConsentArtefact)
+                .signature(signature)
+                .status(ConsentStatus.EXPIRED)
+                .build();
+    }
```

Comment:
Why we are sending the whole consent Artefact to HIP? It just a notification to HIP and HIU that consent (artefact) is expired right?

Reply:
We should be doing same thing for revoke consent, if not that was a mistake.

Reply:
Yes the same thing is done in revoke too. Complete artefact is sent to HIP.

Reply:
modified the contract between consentmanager and hip to include only the necessary details.